src/datasources/email/email.datasource.ts

```diff
@@ -199,4 +221,87 @@ export class EmailDataSource implements IEmailDataSource {
       );
     }
   }
+
+  async getSubscriptions(args: {
+    chainId: string;
+    safeAddress: string;
+    account: string;
+  }): Promise<DomainSubscription[]> {
+    const subscriptions = await this.sql`SELECT key, name
+                                         FROM emails.subscriptions
+                                                  INNER JOIN emails.account_subscriptions subs
+                                                             on subscriptions.id = subs.subscription_id
+                                                  INNER JOIN emails.account_emails emails on emails.id = subs.account_id
+                                         WHERE chain_id = ${args.chainId}
+                                           AND safe_address = ${args.safeAddress}
+                                           AND account = ${args.account}`;
+    return subscriptions.map(
+      (subscription) =>
+        <DomainSubscription>{
+          key: subscription.key,
+          name: subscription.name,
+        },
+    );
+  }
+
+  async subscribe(args: {
+    chainId: string;
+    safeAddress: string;
+    account: string;
+    categoryKey: string;
+  }): Promise<DomainSubscription[]> {
+    const subscriptions = await this
+      .sql`INSERT INTO emails.account_subscriptions (account_id, subscription_id)
```

Comment:
Can you show an example on where you'd like to have the parentheses?

Regarding indentation it is left to prettier. I can change it locally but it is then formatted as you see here.

Reply:
I was thinking in something like this to distinguish a bit more the main query (INSERT) from the subquery (nested SELECT):

```sql
INSERT INTO emails.account_subscriptions (account_id, subscription_id)
    (SELECT account_emails.id AS account_id,
            subscriptions.id  AS subscription_id
    FROM emails.account_emails
            CROSS JOIN emails.subscriptions
    WHERE account_emails.chain_id = ${args.chainId}
    AND account_emails.safe_address = ${args.safeAddress}
    AND account_emails.account = ${args.account}
    AND subscriptions.key = ${args.categoryKey})
RETURNING *`;
```

This would also apply here: https://github.com/safe-global/safe-client-gateway/blob/040d7d6378f9fe26daf364d342d3a9481c361b8a/src/datasources/email/email.datasource.ts#L98 (out of the scope of this PR)

But feel free to ignore this, as we commented in the past since we don't have strict rules regarding SQL formatting yet.

Reply:
With this example in particular I actually don't find that the parentheses help that much because it'd be for most of the query.

However if we talk about indentation then this is something that we should align. It'd be better to have a tool to format SQL queries instead.

Reply:
While I agree it's not critical to include parenthesis in this particular case regarding readability, I do think it's a common/good practice to use them for subqueries (even if they enclose most of the query). I see this practice in all the references I'm checking:
- https://www.postgresql.org/docs/current/functions-subquery.html
- https://www.postgresql.org/docs/current/queries-with.html
- https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/SELECT.html#GUID-CFA006CA-6FF1-4972-821E-6996142A51C6__I2066912
- https://www.oreilly.com/library/view/practical-postgresql/9781449309770/ch04s07.html

But it's fine by me if you think we don't need it. I agree we can defer these discussions until we have rules/a tool to format SQL 👍

Reply:
Sounds good. I don't have a strong opinion but since it improves readability I added the parentheses 👍 Added in [81f95ca](https://github.com/safe-global/safe-client-gateway/pull/1028/commits/81f95cac601715d232425e390d9090640f918d9e)